modules/user-groups/user-groups.php

```diff
@@ -667,571 +722,600 @@ function user_profile_page() {
-			'list_id' => 'ef-following_usergroups',
-			'input_id' => 'following_usergroups'
-		);
-
-		$parsed_args = wp_parse_args( $args, $defaults );
-		extract( $parsed_args, EXTR_SKIP );
-		$usergroups = $this->get_usergroups();
-		if ( empty($usergroups) ) {
-			?>
-			<p><?php _e('No user groups were found.', 'edit-flow') ?> <a href="<?php echo esc_url( $this->get_link() ); ?>" title="<?php _e('Add a new user group. Opens new window.', 'edit-flow' ) ?>" target="_blank"><?php _e( 'Add a User Group', 'edit-flow' ); ?></a></p>
-			<?php
-		} else {
 
-			?>
-			<ul id="<?php echo esc_attr( $list_id ) ?>" class="<?php echo esc_attr( $list_class ) ?>">
-			<?php
-			foreach( $usergroups as $usergroup ) {
-				$checked = ( in_array( $usergroup->term_id, $selected ) ) ? ' checked="checked"' : '';
+		/**
+		 * Displays a list of usergroups with checkboxes
+		 *
+		 * @since 0.7
+		 *
+		 * @param array $selected List of usergroup keys that should be checked
+		 * @param array $args ???
+		 */
+		public function usergroups_select_form( $selected = array(), $args = null ) {
+
+			// TODO add $args for additional options
+			// e.g. showing members assigned to group (John Smith, Jane Doe, and 9 others)
+			// before <tag>, after <tag>, class, id names?
+			$defaults = array(
+				'list_class' => 'ef-post_following_list',
+				'list_id' => 'ef-following_usergroups',
+				'input_id' => 'following_usergroups',
+			);
+
+			$parsed_args = wp_parse_args( $args, $defaults );
+			extract( $parsed_args, EXTR_SKIP );
+			$usergroups = $this->get_usergroups();
+			if ( empty( $usergroups ) ) {
 				?>
+			<p><?php _e( 'No user groups were found.', 'edit-flow' ); ?> <a href="<?php echo esc_url( $this->get_link() ); ?>" title="<?php _e( 'Add a new user group. Opens new window.', 'edit-flow' ); ?>" target="_blank"><?php _e( 'Add a User Group', 'edit-flow' ); ?></a></p>
+				<?php
+			} else {
+
+				?>
+			<ul id="<?php echo esc_attr( $list_id ); ?>" class="<?php echo esc_attr( $list_class ); ?>">
+				<?php
+				foreach ( $usergroups as $usergroup ) {
+					$checked = ( in_array( $usergroup->term_id, $selected ) ) ? ' checked="checked"' : '';
+					?>
 				<li>
-					<label for="<?php echo $input_id . esc_attr( $usergroup->term_id ); ?>" title="<?php echo esc_attr($usergroup->description) ?>">
+					<label for="<?php echo esc_attr( $input_id ) . esc_attr( $usergroup->term_id ); ?>" title="<?php echo esc_attr( $usergroup->description ); ?>">
 						<div class="ef-user-subscribe-actions">
-							<input type="checkbox" id="<?php echo esc_attr( $input_id . $usergroup->term_id ) ?>" name="<?php echo esc_attr( $input_id )?>[]" value="<?php echo esc_attr( $usergroup->term_id ) ?>"<?php echo $checked ?> />
+							<input type="checkbox" id="<?php echo esc_attr( $input_id . $usergroup->term_id ); ?>" name="<?php echo esc_attr( $input_id ); ?>[]" value="<?php echo esc_attr( $usergroup->term_id ); ?>"<?php echo wp_kses_post( $checked ); ?> />
```

Comment:
WordPress has a [built-in `checked()` method](https://developer.wordpress.org/reference/functions/checked/) that should let us avoid using `wp_kses_post()`, and also avoid precomputing the HTML value of `$checked`:

```suggestion
							<input type="checkbox" id="<?php echo esc_attr( $input_id . $usergroup->term_id ); ?>" name="<?php echo esc_attr( $input_id ); ?>[]" value="<?php echo esc_attr( $usergroup->term_id ); ?>"<?php checked( in_array( $usergroup->term_id, $selected ) ); ?> />
```

Reply:
It's interesting because phpcs still raised an error for this. Lemme try it again

Reply:
Done